### PR TITLE
ci(devops): :technologist: add action to assign PR to creator

### PR DESCRIPTION
**Pull request by signekb**
## Description

These changes are so we don't have to do it manually.

Related to rostools/cog-flow-intro#18 but doesn't close it since we need a `sync` workflow as well to add it to the remaining repositories.

## Reviewer Focus

This PR needs an in-depth review.

### CODE REVIEW

File: .github/workflows/assign-pr-to-creator.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    types:
+      - reopened
+      - opened
+
+permissions:
+  pull-requests: write
+
+jobs:
+  assign-pr-to-creator:
+    name: Assign PR to creator
```
Comment (signekb) [inline .github/workflows/assign-pr-to-creator.yml:14]:
```suggestion
```

File: .github/workflows/assign-pr-to-creator.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Assign PR to creator
+        if: ${{ github.event_name == 'pull_request' }}
+        run: |
+          gh pr edit $PR --add-assignee $AUTHOR
+        env:
+          AUTHOR: ${{ github.event.pull_request.user.login }}
+          PR: ${{ github.event.pull_request.html_url }}
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (lwjohnst86) [inline .github/workflows/assign-pr-to-creator.yml:23]:
```suggestion
          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}

```

Git likes an empty line at the end.

File: .github/workflows/assign-pr-to-creator.yml
Comment (lwjohnst86) [inline .github/workflows/assign-pr-to-creator.yml:1]: If we want this in the cog flow, we also would need to set up a synching workflow too.

**Review by lwjohnst86 — CHANGES_REQUESTED** [review]
:tada:

File: .github/workflows/assign-pr-to-creator.yml
Comment (signekb) [reply .github/workflows/assign-pr-to-creator.yml:1]: Ohhh, I don't know how to do that. Maybe we can take a look at it together?

**Review by lwjohnst86 — APPROVED** [review]

**Pull request merged**